### PR TITLE
Improve batch shape handling in MEADS and allow scaled step sizes in GHMC using `mcmc.integrators.velocity_verlet`

### LINKED ISSUE #285: Improve MEADS implementation
- [ ] Improve some shape handling related to batch dimension (e.g., https://github.com/blackjax-devs/blackjax/pull/229/files#r973546362)
- [ ] change the per variable step size to using scalar step size and diag mass matrix

### PR DESCRIPTION
**Pull request by albcab**
Also removing some unused parameters of high-level functions, see #285.

Moved `maximum_eigenvalue` and improved docstring (https://github.com/blackjax-devs/blackjax/pull/229/files#r973546148).

Close #285

### CODE REVIEW

File: blackjax/adaptation/meads.py [hunk 1]
```diff
@@ -9,10 +9,12 @@
 class MEADSAdaptationState(NamedTuple):
     """State of the MEADS adaptation scheme.
 
-    step_sizes
-        The version of the generalized HMC algorithm that is implemented uses a
-        scalar mass matrix, and preconditioning happens at the step size level.
-        There is thus one value of the step size per dimension of the problem.
+    step_size
+        Value of the step_size parameter of the generalized HMC algorithm.
+    sd_position
```
Comment (junpenglao) [inline blackjax/adaptation/meads.py:14]:
```suggestion
    position_sigma
```

Same below

**Review by junpenglao — COMMENTED** [review]
This looks great! Thanks!

File: blackjax/kernels.py [hunk 2]
```diff
@@ -765,9 +765,6 @@ def kernel(rng_key, state):
 def meads(
     logprob_fn: Callable,
     num_chains: int,
-    *,
-    divergence_threshold: int = 1000,
```
Comment (rlouf) [inline blackjax/kernels.py:769]: Why are we dropping the `divergence_threshold` argument?

File: blackjax/mcmc/ghmc.py [hunk 3]
```diff
@@ -152,62 +161,6 @@ def potential_fn(x):
     return one_step
 
 
-def velocity_verlet(
```
Comment (rlouf) [inline blackjax/mcmc/ghmc.py:155]: NICE

**Pull request merged**